### ISSUE TITLE
Wal not working with polybar

### ISSUE THREAD

**Issue by harsha003** [dylanaraps/wal#47]
It works with everything else but not polybar.Polybar directly takes the colors from the ~/.Xresources file . Here's my polybar config:
```
[colors]
background = ${xrdb:background}
foreground = ${xrdb:foreground}
black = ${xrdb:color0}
red = ${xrdb:color1}
green = ${xrdb:color2}
yellow = ${xrdb:color3}
blue = ${xrdb:color4}
magenta = ${xrdb:color5}
cyan = ${xrdb:color6}
white = ${xrdb:color7}







[bar/mybar]


override-redirect = true
width = 100%
height = 25
fixed-center = true
bottom = true
background = ${colors.background}
foreground = ${colors.foreground}
radius = 0.0
line-size = 3
line-color = #f00


padding-left = 0
padding-right = 0

module-margin-left = 2
module-margin-right = 3


module-margin = 1

font-0 = Iosevka-Regular:size=10;2
font-1 = FontAwesome:size9
font-2 = Iosevka Regular:size=9
modules-left = battery i3 
modules-center = mpd 
modules-right =  wireless-network backlight volume date redshift menu0

dim-value =0.0

wm-restack = i3

enable-ipc = true
;;;;;;;;;;;;
;;        ;;
;;MODULES ;;
;;        ;;
;;;;;;;;;;;;

[module/backlight]
type = custom/script
exec = /bin/bash round.sh 1 (light)
interval = 0.1

format-prefix = 
format-prefix-foreground = ${colors.yellow}
format-padding = 1
format-underline = ${colors.yellow}
label = %output%%
label-padding-left = 1

scroll-up = light -A 10
scroll-down = light -U 10

[module/battery]
type = internal/battery
full-at = 99
battery = BAT0
adapter = ACAD

time-format = %H:%M
format-charging = <animation-charging> <label-charging>
format-charging-padding = 1
format-charging-overline = ${colors.green}
format-charging-foreground = ${colors.green}

format-discharging = <ramp-capacity> <label-discharging>
format-full-underline = ${colors.green}
format-full-padding = 1

label-charging = %{F#b7b8b9}%percentage:2%%%{F-} %time%
label-discharging = %percentage:2%% %time% %{o- -o}
label-full = %{F#31a354}%{F-} %percentage%% 

ramp-capacity-0 = %{F#e31a1c o#e31a1c +o} %{F-}
ramp-capacity-1 = %{F#e31a1c o#e31a1c +o} %{F-}
ramp-capacity-2 = %{F#d7a645 o#d7a645 +o} %{F-}
ramp-capacity-3 = %{F#31a354 o#31a354 +o} %{F-}
ramp-capacity-4 = %{F#31a354 o#31a354 +o} %{F-}

animation-charging-0 = 
animation-charging-1 = 
animation-charging-2 = 
animation-charging-3 = 
animation-charging-4 = 

[module/date]
type = internal/date
;date = %d%-%m-%Y%
time = %H:%M
format =  <label>
format-background = ${colors.background}
format-foreground = ${colors.foreground}
label = %time%
label-foreground = ${colors.foreground}
click-left = cal

[module/i3]
type = internal/i3
index-sort = false
strip-wsnumbers = true

ws-icon-0 = 1; 
ws-icon-1 = 2;
ws-icon-2 = 3;
ws-icon-3 = 4;
ws-icon-4 = 5;
ws-icon-5 = 6;6
ws-icon-6 = 7;7
ws-icon-7 = 8;8
ws-icon-8 = 9;9
ws-icon-9 = 10;

format = <label-state>

label-unfocused-background = ${colors.background}
label-mode = %mode%
label-mode-padding = 1
label-mode-background = ${colors.background}

label-focused = %icon%
label-focused-foreground = ${colors.foreground} 
label-focused-background = ${colors.black}
label-focused-underline = #fae3bb
label-focused-padding = 1

label-unfocused = %icon%
label-unfocused-padding = 1 

label-visible = %icon%
label-visible-underline = #555555
label-visible-padding = 1

label-urgent = %icon%
label-urgent-foreground = ${colors.foreground}
label-urgent-background = ${colors.red}
label-urgent-padding = 1

[module/menu0]
type = custom/menu

format = <label-toggle> <menu>
format-underline = ${colors.white}
format-padding = 1

menu-0-0 =  shutdown
menu-0-0-exec = shutdown now
menu-0-1 =  restart
menu-0-1-exec = shutdown -r now
menu-0-2 =  cancel

label-open =   
label-close =


[module/mpd]
type = internal/mpd

host = 127.0.0.1
port = 6600

interval = 1

format-online = <label-song> <bar-progress>
icon-play =  
icon-pause = 
label-song = %title%.%artist%
bar-progress-width = 10
bar-progress-indicator = |
bar-progress-fill = ─
bar-progress-empty = ─

[module/wireless-network]
type = internal/network
interface = wlp3s0
interval = 3.0

format-connected-padding = 1
format-disconnected-padding = 1

format-connected = <label-connected>
format-connected-underline = ${colors.black}
format-disconnected-underline = ${colors.yellow}

label-connected = %essid%
label-disconnected = 
label-disconencted-foreground = ${colors.red}



[module/volume]
type = custom/script
exec = volume.sh
label = %output%
interval = 0.1
scroll-up = pamixer --sink 1 -i 5
scroll-down = pamixer --sink 1 -d 5

[module/redshift]
type = custom/script
interval = 5
#format-prefix = 
format-prefix-foreground = ${colors.yellow}
format-padding = 1
format-underline = ${colors.yellow}
label = %output%
label-padding-left = 1
exec = ~/.config/polybar/redshift.sh
```
Here's what it looks like:
[Bar](http://i.imgur.com/diaWlm2.png)

[Colors](http://i.imgur.com/hbsWkwS.png)

**Comment by RorixRebel** [comment]
Whats your output for xrdb -query?
it should be the 15+ colors that reside in your /.xresources

**Comment by harsha003** [comment]
here it is : http://ix.io/wBg

**Comment by RorixRebel** [comment]
your colors appear on the list twice under the same name.
not sure thats the problem but im guessing your polybar doesnt know which one to pick.

`*.background:	#002b36
*.color0:	#002b36
*.color1:	#dc322f
*.color10:	#859900
*.color11:	#b58900
*.color12:	#268bd2
*.color13:	#6c71c4
*.color14:	#2aa198
*.color15:	#fdf6e3
*.color2:	#859900
*.color3:	#b58900
*.color4:	#268bd2
*.color5:	#6c71c4
*.color6:	#2aa198
*.color7:	#93a1a1
*.color8:	#657b83
*.color9:	#dc322f
*.cursorColor:	#93a1a1
*.foreground:	#93a1a1
*antialias:	true
*autohint:	true

*color0:	#04080C
*color1:	#074C34
*color10:	#0E602E
*color11:	#076B4B
*color12:	#17B753
*color13:	#0A9B62
*color14:	#A69167
*color15:	#4DC48F
*color2:	#0E602E
*color3:	#076B4B
*color4:	#17B753
*color5:	#0A9B62
*color6:	#A69167
*color66:	#04080C
*color7:	#4DC48F
*color8:	#666666
*color9:	#074C34`

**Comment by harsha003** [comment]
Is there a way to remove one?

**Comment by RorixRebel** [comment]
give it a go, as long as you only have 1 set of colors (1-15)

**Comment by harsha003** [comment]
It reverted back to the previous two-schemed output.

**Comment by RorixRebel** [comment]
there has to be something in there that is already populating your xresources

**Comment by harsha003** [comment]
I found this code in the wal script 
```
reload_xrdb() {
    [[ "$alpha" ]] && x_colors+="URxvt.depth: 32${newline}"

    # Merge the colors into the X db so new terminals use them.
    xrdb -merge >/dev/null 2>&1 <<< "$x_colors" && \
        out "colors: Merged colors with X env"
}

```
I'm quite new to linux,so I don't know a lot about shell scripts.
Is this the reason?

**Comment by HonorOfDuty** [comment]
I have the same problem, for some reason wal generates a color scheme with an extra color `.color66:` which throws the entire xrdb, and probably therfore polybar off. I whish i knew something bout code so that i maybe was able to fix this but alas.

**Comment by dylanaraps** [comment]
The `color66` shouldn't confuse or break xrdb. The `color66` is only there for vim airline as bold highlighting breaks the black text color. (It uses color 8 instead).

```sh
    # Set a black color that isn't affected by bold highlighting.
    sequences+="\033]4;66;${colors[0]}\007"
    x_colors+="*color66: ${colors[0]}${newline}"
```

**Comment by harsha003** [comment]
So it is not wal's fault then?

**Comment by RorixRebel** [comment]
Shouldnt, there's something in your systems generating data and dumping it on xresources causing duplicate data.

Could i suggest an "apt list --installed" to see what yall have installed. Cause it's 2 guys now.

**Comment by harsha003** [comment]
So yesterday I reinstalled polybar and now the colors are working fine.Maybe something got corrupted.

**Comment by IOffendedSomeoneAndHadToChangeMyName** [comment]
Could you xrdb -query again please? Re-installing did nothing for me and I also have two sets of colors in there.

**Comment by HonorOfDuty** [comment]
Managed to fix it. Before I started to use wal, I sued URXVTconfig to generate colorschemes for my terminal from my current backgrounds. This left it's colorschemes in both Xresources and Xdefaults.

What I did was uninstall URXVTconfig, delete the colors from my .Xresources and .Xdefaults (just in case), reload/rerun `wal`, run `xrdb -query` to see if there is now only one colorsheme in Xresources, and last but not least, restart polybar with the launch script.
